oauthenticator/oauth2.py

```diff
@@ -663,6 +681,19 @@ def _allowed_scopes_validation(self, proposal):
         """,
     )
 
+    pkce = Bool(
+        os.environ.get("OAUTH2_PKCE", "False").lower() in {"true", "1"},
+        config=True,
+        help="""
+        Whether to use PKCE (Proof Key for Code Exchange) for the OAuth2 flow.
+
+        When using PKCE, the client secret is not required to be sent to the
+        token endpoint.
```

Comment:
https://www.oauth.com/oauth2-servers/pkce/authorization-request/ says

> The authorization server can require that public clients must use the PKCE extension. This is really the only way to allow native apps to have a secure authorization flow without using the client secret, especially without the redirect URI security that’s available with web-based clients.

Does this imply it should still be sent unless the server requires PKCE?

Reply:
You're right. In fact, the code itself was contradicting this comment. I removed the part about `client_secret`.